DatadogRUM/Tests/RUMMonitor/Monitor+GlobalAttributesTests.swift

```diff
@@ -0,0 +1,419 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+import TestUtilities
+import DatadogInternal
+@testable import DatadogRUM
+
+class Monitor_GlobalAttributesTests: XCTestCase {
+    private let featureScope = FeatureScopeMock()
+    private var monitor: Monitor! // swiftlint:disable:this implicitly_unwrapped_optional
+
+    override func setUp() {
+        monitor = Monitor(
+            dependencies: .mockWith(featureScope: featureScope),
+            dateProvider: SystemDateProvider()
+        )
+    }
+
+    override func tearDown() {
+        monitor = nil
+    }
+
+    // MARK: - Changing Global Attributes After SDK Init
+
+    func testAddingGlobalAttributeAfterSDKInit() throws {
+        // When
+        monitor.notifySDKInit()
+        monitor.addAttribute(forKey: "attribute", value: "value")
+
+        // Then
+        let lastView = try XCTUnwrap(featureScope.eventsWritten(ofType: RUMViewEvent.self).last)
+        XCTAssertEqual(lastView.view.name, RUMOffViewEventsHandlingRule.Constants.applicationLaunchViewName)
+        XCTAssertNil(lastView.attribute(forKey: "attribute"))
+    }
+
+    func testAddingGlobalAttributeAfterSDKInit_thenStartingView() throws {
+        // Given
+        monitor.notifySDKInit()
+        monitor.addAttribute(forKey: "attribute", value: "value")
+
+        // When
+        monitor.startView(key: "View")
+
+        // Then
+        let viewEvents = featureScope.eventsWritten(ofType: RUMViewEvent.self)
+        let appLaunchView = try XCTUnwrap(viewEvents.last(where: { $0.view.name == RUMOffViewEventsHandlingRule.Constants.applicationLaunchViewName }))
+        XCTAssertEqual(appLaunchView.attribute(forKey: "attribute"), "value")
+    }
+
+    func testAddingGlobalAttributeAfterSDKInit_thenSendingInteractiveEvent() throws {
+        // Given
+        monitor.notifySDKInit()
+        monitor.addAttribute(forKey: "attribute", value: "value")
+
+        // When
+        monitor.addAction(type: .custom, name: "custom action")
+
+        // Then
+        let lastView = try XCTUnwrap(featureScope.eventsWritten(ofType: RUMViewEvent.self).last)
+        XCTAssertEqual(lastView.view.name, RUMOffViewEventsHandlingRule.Constants.applicationLaunchViewName)
+        XCTAssertNil(lastView.attribute(forKey: "attribute"))
+    }
+
+    func testAddingGlobalAttributesAfterSDKInit_thenRemovingAttribute() throws {
+        // Given
+        monitor.notifySDKInit()
+        monitor.addAttribute(forKey: "attribute1", value: "value1")
+        monitor.addAttribute(forKey: "attribute2", value: "value2")
+
+        // When
+        monitor.removeAttribute(forKey: "attribute1")
```

Comment:
/suggestion

- can we add a test something like removing an attribute which doesn't exist? Ideally should be a noop.

- also adding the same key attribute again

Reply:
Fair point 👍. Added extra coverage for updating attributes and removing non-existing ones.